plugins/woocommerce/tests/e2e-pw/tests/smoke-tests/update-woocommerce.spec.js

```diff
@@ -20,34 +20,67 @@ const skipTestIfUndefined = () => {
 	}, skipMessage );
 };
 
+/**
+ *
+ * Get download URL of a WooCommerce ZIP asset by sending a `GET` request to `List releases` GitHub API endpoint in the WooCommerce repository.
+ *
+ * If `GITHUB_TOKEN` is defined, use it as `Authorization` header.
+ *
+ * @returns Download URL of the WooCommerce ZIP. This URL depends on whether `GITHUB_TOKEN` was specified or not.
+ *
+ * If `GITHUB_TOKEN` was defined, this function assumes that you're trying to access all releases, including drafts (as draft releases don't show up in the response of an unauthenticated GET `List releases` request ).
+ * In this case, the returned value will be the `asset.url`.
+ *
+ * Otherwise, the returned value will be the `asset.browser_download_url`.
+ *
+ * @throws Error if:
+ * - 'List releases' request was unsuccessful, or
+ * - no release with the given tag was found, or
+ * - when a WooCommerce ZIP asset was not found.
+ *
+ */
 const getWCDownloadURL = async () => {
-	let woocommerceZipAsset;
-
-	const response = await axios
-		.get(
-			`https://api.github.com/repos/woocommerce/woocommerce/releases/tags/${ UPDATE_WC }`
-		)
-		.catch( ( error ) => {
-			console.log( error.toJSON() );
-			return error.response;
-		} );
+	const requestConfig = {
+		method: 'get',
+		url: 'https://api.github.com/repos/woocommerce/woocommerce/releases',
+		headers: {
+			Accept: 'application/vnd.github+json',
+		},
+	};
+
+	if ( GITHUB_TOKEN ) {
+		requestConfig.headers.Authorization = `Bearer ${ GITHUB_TOKEN }`;
+	}
```

Comment:
The changes in this `update-woocommerce.spec.js` file centers around this particular line. This makes sure that the supplied token will be used when downloading the `woocommerce.zip` file from a release, whether if it's only a draft one or already published.